File: frontend/src/types/session.ts
import type { StrokePoint } from './stroke';

export interface SessionSummary {
  id: string;
  name: string;
  created_at: string;
  updated_at: string;
  stroke_count: number;
}

export interface StrokeRecord {
  id: string;
  session_id: string;
  points: StrokePoint[];
  pen_color: string;
  pen_width: number;
  created_at: string;
}

export interface SessionDetail {
  id: string;
  name: string;
  created_at: string;
  updated_at: string;
  strokes: StrokeRecord[];
}

// Shape sent to POST /api/sessions/{id}/strokes — matches backend StrokeCreate schema
export interface StrokeCreatePayload {
  id: string;
  points: StrokePoint[];
  pen_color: string;
  pen_width: number;
}

File: backend/app/schemas/session.py
from datetime import datetime

from pydantic import BaseModel

from app.schemas.stroke import StrokeResponse


class SessionCreate(BaseModel):
    name: str = ""


class SessionSummary(BaseModel):
    id: str
    name: str
    created_at: datetime
    updated_at: datetime
    stroke_count: int


class SessionDetail(BaseModel):
    id: str
    name: str
    created_at: datetime
    updated_at: datetime
    strokes: list[StrokeResponse]
    transcript: list[str] = []
